dockerclient.go

```diff
@@ -231,6 +231,48 @@ func (client *DockerClient) ContainerChanges(id string) ([]*ContainerChanges, er
 	return changes, nil
 }
 
+func (client *DockerClient) readJSONStream(stream io.ReadCloser, decode func(*json.Decoder) decodingResult, stopChan <-chan struct{}) <-chan decodingResult {
+	resultChan := make(chan decodingResult)
+	internalResultsChan := make(chan decodingResult)
+	stillListening := make(chan struct{})
+
+	go func() {
+		decoder := json.NewDecoder(stream)
+		defer stream.Close()
+		for {
+			decodeResult := decode(decoder)
+			if _, ok := <-stillListening; !ok {
+				return
+			}
+			internalResultsChan <- decodeResult
+			if decodeResult.err != nil {
+				return
+			}
+		}
+	}()
+
+	go func() {
```

Comment:
Sorry, I wasn't clear :) I think this second goroutine can be removed altogether